src/Facets/AcrossFacetPacked.sol

```diff
@@ -0,0 +1,298 @@
+        uint256 maxCount,
+        bytes calldata message
+    ) external pure returns (bytes memory) {
+        // there are already existing networks with chainIds outside uint32 range but since we not support either of them yet,
+        // we feel comfortable using this approach to save further gas
+        require(
+            destinationChainId <= type(uint32).max,
+            "destinationChainId value passed too big to fit in uint32"
+        );
+
+        return
+            bytes.concat(
+                AcrossFacetPacked
+                    .startBridgeTokensViaAcrossNativePacked
+                    .selector,
+                bytes8(transactionId),
+                bytes20(receiver),
+                bytes4(uint32(destinationChainId)),
+                bytes8(uint64(relayerFeePct)),
+                bytes4(quoteTimestamp),
+                bytes32(maxCount),
+                message
+            );
+    }
+
+    function encode_startBridgeTokensViaAcrossERC20Packed(
+        bytes32 transactionId,
+        address receiver,
+        address sendingAssetId,
+        uint256 minAmount,
+        uint64 destinationChainId,
+        int64 relayerFeePct,
+        uint32 quoteTimestamp,
+        bytes calldata message,
+        uint256 maxCount
+    ) external pure returns (bytes memory) {
+        // there are already existing networks with chainIds outside uint32 range but since we not support either of them yet,
+        // we feel comfortable using this approach to save further gas
+        require(
+            destinationChainId <= type(uint32).max,
+            "destinationChainId value passed too big to fit in uint32"
+        );
+
+        require(
+            minAmount <= type(uint128).max,
+            "minAmount value passed too big to fit in uint128"
+        );
+
+        return
+            bytes.concat(
+                AcrossFacetPacked
+                    .startBridgeTokensViaAcrossERC20Packed
+                    .selector,
+                bytes8(transactionId),
+                bytes20(receiver),
+                bytes20(sendingAssetId),
+                bytes16(uint128(minAmount)),
+                bytes4(uint32(destinationChainId)),
+                bytes8(uint64(relayerFeePct)),
+                bytes4(uint32(quoteTimestamp)),
+                bytes32(maxCount),
+                message
+            );
+    }
+
+    function decode_startBridgeTokensViaAcrossNativePacked(
+        bytes calldata data
+    )
+        external
+        pure
+        returns (
+            BridgeData memory bridgeData,
+            AcrossFacet.AcrossData memory acrossData
+        )
+    {
+        require(data.length >= 80, "invalid calldata (must have length > 80)");
+
+        // extract bridgeData
+        bridgeData.transactionId = bytes32(bytes8(data[4:12]));
+        bridgeData.receiver = address(bytes20(data[12:32]));
+        bridgeData.destinationChainId = uint64(uint32(bytes4(data[32:36])));
```

Comment:
for consistency here as well. I guess currently it is casted automatically as the `bridgeData` interface uses `uint256`

Reply:
yes correct, it's casted automatically